src/publish.proj

```diff
@@ -1,41 +1,52 @@
-        <RelativeBlobPath Condition="'$(PublishTestNativeBins)' == 'true'">$(__BuildType)/%(RecursiveDir)%(Filename)%(Extension)</RelativeBlobPath>    
-      </ForPublishing>
-    </ItemGroup>
-    <Error Condition="'@(ForPublishing)' == ''" Text="No items were found matching pattern '$(PublishPattern)'." />
-  </Target>
+  <PropertyGroup>
+    <FeedsDir Condition="'$(BuildOS)' == 'Windows_NT'">$(MSBuildThisFileDirectory)../component/Microsoft.Dotnet.Build.Tasks.Feed/desktop/</FeedsDir>
+    <FeedsDir Condition="'$(BuildOS)' != 'Windows_NT'">$(MSBuildThisFileDirectory)../component/</FeedsDir>
+  </PropertyGroup>
+
+  <Import Project="$(FeedsDir)Publish.targets" />
+
+  <PropertyGroup>
+    <PackagesPattern Condition="'$(PackagesPattern)' == '' and '$(PublishTestNativeBins)' != 'true'">$(PackagesBinDir)pkg\*.nupkg</PackagesPattern>
+    <PackagesPattern Condition="'$(PackagesPattern)' == '' and '$(PublishTestNativeBins)' == 'true'">$(OutputPath)\bin\**</PackagesPattern>
+    <SymbolsPackagesPattern>$(PackagesBinDir)symbolpkg\*.nupkg</SymbolsPackagesPattern>
+  </PropertyGroup>
 
   <PropertyGroup>
-    <PublishPattern Condition="'$(PublishPattern)' == '' and '$(PublishTestNativeBins)' != 'true'">$(PackagesBinDir)**\*.nupkg</PublishPattern>
-    <PublishPattern Condition="'$(PublishPattern)' == '' and '$(PublishTestNativeBins)' == 'true'">$(OutputPath)\bin\**</PublishPattern>
+    <ContainerName Condition="'$(__Container)' == '' and '$(PublishTestNativeBins)' != 'true'">coreclr-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)</ContainerName>
+    <ContainerName Condition="'$(__Container)' == '' and '$(PublishTestNativeBins)' == 'true'">coreclr-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)-$(__DistroRid)-$(__BuildArch)</ContainerName>
+    <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' != 'true'">$(__Container)</ContainerName>
+    <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' == 'true'">$(__Container)-$(__DistroRid)-$(__BuildArch)</ContainerName>
+    <ContainerName>$(ContainerName.Replace(".","-"))</ContainerName>
+    <AccountName>$(CloudDropAccountName)</AccountName>
+    <AccountKey>$(CloudDropAccessToken)</AccountKey>
   </PropertyGroup>
 
-  <Target Name="CreateContainerName"
-          DependsOnTargets="CreateVersionFileDuringBuild"
-          Condition="'$(ContainerName)' == '' or '$(PublishTestNativeBins)' == 'true'">
+  <Target Name="PublishPackages" BeforeTargets="PublishOutputLeg" Condition="'$(__PublishPackages)' == 'true'">
+    <PropertyGroup>
+      <FeedType>Blob</FeedType>
+      <RelativePath>packages</RelativePath>
+    </PropertyGroup>
+    <ItemGroup>
+      <ItemsToPush Remove="*.nupkg" />
+      <ItemsToPush Include="$(PackagesPattern)" Exclude="$(SymbolsPackagesPattern)"/>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="PublishSymbolPackages" BeforeTargets="PublishOutputLeg"  Condition="'$(__PublishSymbols)' == 'true'" >
```

Comment:
Instead of doing a BeforeTargets PublishOutputLeg we should directly call the PushToBlobFeed task https://github.com/dotnet/buildtools/blob/57ac79a55928b9693ed3973ef068d75d490d4c5c/src/Microsoft.DotNet.Build.Tasks.Feed/PackageFiles/Publish.targets#L24. I think that makes it a lot clearer of what is happening.

Reply:
What's the syntax for this? I tried invoking the target directly, but then PropertyGroups weren't passed through; I also tried msbuild-ing that target file, but then I couldn't pass through ItemGroups. This was the only way I found to get both plumbed through.

Reply:
Don't call the Target but the task. Just like in the target that I linked to is doing.

Reply:
Ah, I see. What is the point of that target then? Does it get used anywhere else?

Reply:
I don't see much point in it personally beyond serving as an example.